draft-ietf-httpbis-http2bis.xml

```diff
@@ -3216,9 +3215,23 @@
         <t>
           HTTP/2 allows a server to pre-emptively send (or "push") responses (along with
           corresponding "promised" requests) to a client in association with a previous
-          client-initiated request. This can be useful when the server knows the client will need
-          to have those responses available in order to fully process the response to the original
-          request.
+          client-initiated request.
+        </t>
+        <t>
+          Server push was designed to allow a server to improve client-perceived performance by
+          predicting what requests will follow those that it receives, thereby removing a round
+          trip for them. For example, a request for HTML is often followed by requests
+          for stylesheets and scripts associated with that page. When these requests
+          are pushed, the client does not need to wait to receive the references to them in the HTML
+          and issue separate requests.
+        </t>
+        <t>
+          In practice, server push is difficult to use effectively, because it requires the
+          server to correctly anticipate the additional requests the client will make, taking into
+          account factors such as caching, content negotiation, and user behavior. Errors in
+          prediction can lead to performance degradation, due to the opportunity cost that the
+          additional data on the wire represents. In particular, pushing more than a round trip of
+          data can cause contention issues with more-important responses.
```

Comment:
I think that you might say "pushing any significant amount of data".  It doesn't need to be a whole round trip worth.  Even a fraction of a round trip worth can starve out a high priority request that arrives immediately after starting.